### PR TITLE
8344223: Remove calls to SecurityManager and doPrivileged in java.net.URLClassLoader after JEP 486 integration

### PR DESCRIPTION
**Pull request by jaikiran**
Can I please get a review of this change which proposes to remove SecurityManager related API usages from `URLClassLoader` and its related `URLClassPath`? This addresses https://bugs.openjdk.org/browse/JDK-8344223.

The `URLClassLoader.getPermissions()` method will need additional changes but that will be done as part of https://bugs.openjdk.org/browse/JDK-8343150.

I'll be opening a CSR for this change because we are removing support for the `jdk.net.URLClassPath.disableRestrictedPermissions` system property.

No new tests have been added and existing tier1, tier2 and tier3 tests continue to pass.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8344543](https://bugs.openjdk.org/browse/JDK-8344543) to be approved

### Issues
 * [JDK-8344223](https://bugs.openjdk.org/browse/JDK-8344223): Remove calls to SecurityManager and doPrivileged in java.net.URLClassLoader after JEP 486 integration (**Sub-task** - P4)
 * [JDK-8344543](https://bugs.openjdk.org/browse/JDK-8344543): Remove support for jdk.net.URLClassPath.disableRestrictedPermissions after JEP 486 integration (**CSR**)

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Yudi Zheng](https://openjdk.org/census#yzheng) (@mur47x111 - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22233/head:pull/22233` \
`$ git checkout pull/22233`

Update a local copy of the PR: \
`$ git checkout pull/22233` \
`$ git pull https://git.openjdk.org/jdk.git pull/22233/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22233`

View PR using the GUI difftool: \
`$ git pr show -t 22233`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22233.diff">https://git.openjdk.org/jdk/pull/22233.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22233#issuecomment-2485337652)
</details>

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/loader/BuiltinClassLoader.java [hunk 1]
```diff
@@ -355,38 +355,38 @@ public URL findResource(String name) {
...
                 }
             }
         }
 
-        // class path (not checked)
+        // class path
         Enumeration<URL> e = findResourcesOnClassPath(name);
 
-        // concat the checked URLs and the (not checked) class path
+        // concat the resources from the modules and the class path
```
Comment (eirbjo) [inline src/java.base/share/classes/jdk/internal/loader/BuiltinClassLoader.java:387]: Observation: This is now essentially `ClassLoader.CompoundEnumeration`, which could be reused if moved to `jdk.internal.loader`. Probably not for this PR though.

Comment (AlanBateman) [reply src/java.base/share/classes/jdk/internal/loader/BuiltinClassLoader.java:387]: The new comment isn't quite right, it should say "concat the URLs to the resource in the modules and the class path".

**Review by AlanBateman — COMMENTED** [review]
I did a pass over this, good cleanup.

File: src/java.base/share/classes/jdk/internal/loader/BuiltinClassLoader.java (hunk 1 above)
Comment (jaikiran) [reply src/java.base/share/classes/jdk/internal/loader/BuiltinClassLoader.java:387]: > concat the URLs to the resource in the modules and the class path

Did you mean "concat the URLs of the ...." ?

Comment (AlanBateman) [reply src/java.base/share/classes/jdk/internal/loader/BuiltinClassLoader.java:387]: "of" or "to" will work here.

Comment (jaikiran) [reply src/java.base/share/classes/jdk/internal/loader/BuiltinClassLoader.java]: Done - I've now updated this comment.

**Comment by jaikiran** [comment]
The CSR for this change is now ready for review https://bugs.openjdk.org/browse/JDK-8344543

**Comment by jaikiran** [comment]
The CSR for this has been approved and the latest state of this PR continues to pass tier1 and tier2 testing.

**Review by AlanBateman — APPROVED** [review]

**Review by dfuch — COMMENTED** [review]
Not an expert here but those changes look good to me. It's good to see all this complicated code go away!

**Review by mur47x111 — APPROVED** [review]

**Comment by jaikiran** [comment]
Thank you all for the reviews.

**Pull request closed without merging**